### ISSUE TITLE
Add case-insensitive header field access in http request_header

### ISSUE THREAD

**Issue by jbru** [actor-framework/actor-framework#1554]
According to RFC 9110 header field names are case-insensitive: https://www.rfc-editor.org/rfc/rfc9110.html#section-5.1-3

However the `request_header` class only allows case-sensitive access of header fields via the key in https://github.com/actor-framework/actor-framework/blob/master/libcaf_net/caf/net/http/request_header.hpp.

Currently the only way to search for case-insensitive keys is to either compare each key with `request_header::for_each_field()` (but then you can't return early when the field was found) or iterate manually over each field with `request_header::num_fields()` and `request_header::field_at()`.

**Comment by Neverlord** [comment]
Thanks for bringing this up! The RFC is pretty clear on this:

> Field names are case-insensitive

We'll fix it.

**Comment by jbru** [comment]
After writing this I noticed you can't even use `request_header::field_at()` to iterate with an early return from `caf::net::http::responder::header()`, as this returns a `const request_header&` and `request_header::field_at()` is not marked as `const`.

After checking the implementation, the `field_at()` function should be `const`, as it returns by copy:
```
  std::pair<std::string_view, std::string_view> field_at(size_t index) {
    return fields_.container().at(index);
  }
```

**Comment by Neverlord** [comment]
Yeah, that looks like an oversight. Thanks again. 👍

**Comment by Neverlord** [comment]
Just noted: this has been included in 0.19.4 even though it isn't mentioned in the changelog.
